pkg/api/agentPoolOnlyApi/vlabs/validate.go

```diff
@@ -29,9 +29,6 @@ func (a *AgentPoolProfile) Validate() error {
 func (l *LinuxProfile) Validate() error {
 	// Don't need to call validate.Struct(l)
 	// It is handled by Properties.Validate()
-	if e := validate.Var(l.SSH.PublicKeys[0].KeyData, "required"); e != nil {
```

Comment:
same here

Reply:
see previous

Reply:
this is for `agentpoolonly` cluster A.K.A AKS so we definitely want to keep that one.